### PR TITLE
[Woo POS] Tests for classes in com.woocommerce.android.ui.woopos.home dir

### LINKED ISSUE #11932: [Woo POS] Tests for classes in home dir
```[tasklist]
### Tasks
- [x] WooPosCartRepository
- [x] WooPosCartViewModel
- [x] WooPosProductsViewModel
- [x] WooPosTotalsViewModel
- [x] WooPosHomeViewModel
```

### PR DESCRIPTION
**Pull request by backwardstruck**
Closes: #11932

### Description  

Enhances the existing test coverage for the some POS classes by introducing additional test cases to handle various edge cases and interactions with dependencies. 

### Steps to reproduce  

Unit tests should make sense and pass
Code coverage should go up!

#### Devices Used:
- N/A (Unit tests do not require specific devices)

- [X] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/225">19.5</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModelTest.kt [hunk 1]
```diff
@@ -229,6 +232,54 @@ class WooPosCartViewModelTest : BaseUnitTest() {
...
+
+            val sut = createSut()
+            val states = sut.state.captureValues()
+
+            // WHEN
+            parentToChildrenEventsMutableFlow.emit(ParentToChildrenEvent.ItemClickedInProductSelector(1L))
+            sut.onUIEvent(WooPosCartUIEvent.CheckoutClicked)
+
+            // THEN
+            assertEquals(false, states.last().isOrderCreationInProgress)
```
Comment (kidinov) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModelTest.kt]: Most/all the tests use assertJ (`assertThat...`), so maybe we use it in this test class just for consistency?

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartCartRepositoryTest.kt [hunk 2]
```diff
@@ -85,4 +85,13 @@ class WooPosCartCartRepositoryTest {
         assertThat(orderCapture.lastValue.items.size).isEqualTo(3)
         assertThat(orderCapture.lastValue.items.map { it.quantity }).containsExactly(2f, 1f, 3f)
     }
+
+    @Test(expected = IllegalStateException::class)
```
Comment (kidinov) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartCartRepositoryTest.kt]: np: You could replace that with

```
assertFailsWith<IllegalStateException> {
            repository.createOrderWithProducts(productIds)
        }
```

To be specific what function throws the exception

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    @Test
+    fun `initial state is loading`() = runTest {
+        // WHEN
+        val savedState = createMockSavedStateHandle()
+        val viewModel = createViewModel(savedState)
+
+        // THEN
+        assertEquals(WooPosTotalsState.Loading, viewModel.state.value)
```
Comment (kidinov) [inline WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/totals/WooPosTotalsViewModelTest.kt]: Use assertJ here too maybe?

**Review by kidinov — APPROVED** [review]
I left a couple of non blocking comments, but overall probably nice to use the same assertion library everywhere, so we don't create different approaches/styles from the beginning

File: WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModelTest.kt (hunk 1 above)
Comment (backwardstruck) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModelTest.kt]: Sorry I had a note about this and forgot about it. Will update these!

Comment (backwardstruck) [reply WooCommerce/src/test/kotlin/com/woocommerce/android/ui/woopos/home/cart/WooPosCartViewModelTest.kt]: Updated.
